src/steinertree/metric_steiner_tree.jl

```diff
@@ -0,0 +1,49 @@
+"""
+	metric_steiner_tree(g, distmx, term_vert)
+
+`g` is a complete undirected graph.
+Its positive edge weights represented by `distmx` and obeys the triangular inequality.
+`term_vert` represent the terminal vertices of the steiner tree. 
+Finds a tree spanning `g` of minimum weight that connects all the terminal vertices.
+
+### Notes
+Triangular inequality: distmx[a, c] <= distmx[a, b] + distmx[b, c] for all vertices in `g`.
+
+### Input Constraints
+Assumes `g` has an edge for any 2 distinct vertices. 
+distmx obeys the triangle inequality.
+
+Let t = `|term_vert|`
+### Approximation Factor
+2-2/t
+
+### Runtime
+O(t^2 * log(t))
+
+### Implementation Notes
+Perform [Approximate Metric Steiner Tree](lucatrevisan.wordpress.com/2011/01/13/cs261-lecture-2-steiner-tree-approximation/).
+"""
+function metric_steiner_tree(
+	g::AbstractGraph{T},
+	term_vert::Vector{<:Integer},
```

Comment:
I did that to make it more flexible for the user.
Otherwise, `metric_steiner_tree(SimpleGraph{UInt8}(5), [1, 2] )` will result in an error because [1, 2] is of type Vector{Int64}

Reply:
How about I replace
`term_to_actual = unique(term_vert)`
with
`term_to_actual = T.(unique(term_vert))`